packaging/resource_suite_s3_nocache.py

```diff
@@ -1910,6 +1910,39 @@ def test_istream_nonexistent_file__issue_6479(self):
                 'DATA_REPL_STATUS', '0'])
             self.user0.run_icommand(['irm', '-f', logical_path])
 
+    def test_istream_append__issue_2153(self):
+        replica_number_in_s3 = 0
+        filename = 'test_istream_append__issue_2153.txt'
+        logical_path = os.path.join(self.user0.session_collection, filename)
+        content = 'this is the original file for istream'
+        content_append1 = 'line1 appended to the file'
+        content_append2 = 'line2 appended to the file'
+
+        try:
+            # istream to a new data object and ensure that it is created successfully.
+            self.user0.assert_icommand(['istream', 'write', logical_path], input=content)
+            self.assertEqual(str(1), lib.get_replica_status(self.user0, filename, replica_number_in_s3))
+
+            # append to the file
+            self.user0.assert_icommand(['istream', 'write', '-a', logical_path], input=f'\n{content_append1}')
+
+            # append to the file again
+            self.user0.assert_icommand(['istream', 'write', '-a', logical_path], input=f'\n{content_append2}')
+
+            # verify the file size in ils -l - added two newlines above thus the +2 below
+            self.user0.assert_icommand(['ils', '-l', logical_path], 'STDOUT', str(len(content) + len(content_append1) + len(content_append2)+ 2)) 
+
+            # Ensure that the replica actually contains the contents streamed into it.
+            self.user0.assert_icommand(['iget', logical_path, '-'], 'STDOUT_MULTILINE', [content, content_append1, content_append2])
+
+        finally:
+            # Set the replica status here so that we can remove the object even if it is stuck in the locked status.
+            self.admin.run_icommand([
+                'iadmin', 'modrepl',
+                'logical_path', logical_path,
+                'replica_number', str(replica_number_in_s3),
+                'DATA_REPL_STATUS', '0'])
```

Comment:
I feel strongly that we should clean up the data object, and this is the way to do it. The test is reading and writing the data object so any problems will `raise` an `AssertionError`, so changing the replica status in order to clean up will not mask the fact that something went wrong above.

At best, not removing the object will allow the person running the test to inspect the object later, which wouldn't work in an ephemeral environment (e.g. k8s, no `--leak-containers` in testing environment, etc.) and will cause other tests to fail that may have otherwise passed. 

If we want to see more information on failure, the same thing can be accomplished by `print`ing an `ils -L` on the logical path or parent collection right before we do this as a debugging effort. That would work better in an ephemeral environment (i.e. shows up in the logs) and allows other tests to run.

Reply:
I agree with all of that, but if things are working as intended, `iadmin modrepl` isn't needed.

I can see this going both ways. Let's find a few minutes to discuss today.

Reply:
I made the change to check the repl status so there is no chance for this failure to be undetected.  I kept the modrepl so in case there was a failure it doesn't cascade to create a lot of false positives.

Reply:
Sounds good.